platform/android/java/lib/src/org/godotengine/godot/FilePicker.kt

```diff
@@ -0,0 +1,134 @@
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+package org.godotengine.godot
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.provider.DocumentsContract
+import android.util.Log
+import androidx.annotation.RequiresApi
+import org.godotengine.godot.io.file.MediaStoreData
+
+class FilePicker {
+	companion object {
+		private const val FILE_PICKER_REQUEST = 1000
+		private val TAG = FilePicker::class.java.simpleName
+
+		// Constants for fileMode values
+		private const val FILE_MODE_OPEN_FILE = 0
+		private const val FILE_MODE_OPEN_FILES = 1
+		private const val FILE_MODE_OPEN_DIR = 2
+		private const val FILE_MODE_OPEN_ANY = 3
+		private const val FILE_MODE_SAVE_FILE = 4
+
+		@RequiresApi(Build.VERSION_CODES.Q)
+		fun handleActivityResult(context: Context, requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
Could either of those methods be validated using unit tests?

Reply:
I don't think it's feasible, and even if it were possible, it would be challenging due to dependencies on Android-specific components. I also don't have much experience with unit testing, so I'm not entirely sure how to approach testing for this.